### ISSUE TITLE
Notification not appearing on Android after the app is closed/terminated on app first run.

### ISSUE THREAD

**Issue by c0ff33-b34n** [thudugala/Plugin.LocalNotification#246]
When I schedule a notification on an app's first run and terminate the app before the notification is due to appear the notification is not shown. If I leave the app open on first run after setting the notification, the notification is shown. 

Setting a notification on further app runs and terminating the app works fine. But it is on first app run it does not work if the app is terminated. If I install the app, run it and then terminate it (without setting a notification), then launch again and set a notification the notification works.

**To Reproduce**
Schedule a notification during first run of the app (after install).
Swipe away the app to terminate before notification is due.

**Expected behavior**
Notification should appear at time scheduled

**Platform**
 - OS: Android
 - Compiled with Target Android 12.0

**Smartphone**
Google Pixel 2 (emulated) - Android 9.0 - API28.
Samsung Galaxy A32 Phone - Android 11.0 - API 30.

**Code**

In the Android project in MainActivity.cs inside OnCreate():
`
NotificationCenter.CreateNotificationChannel();`

In the shared project I set the notification like so:
			
```
var notification = new NotificationRequest
        {
            NotificationId = 100,
            CategoryType = NotificationCategoryType.Reminder,
            Subtitle = null,
            Title = "Time For A Notification",
            Android =
            {
                IconSmallName =
                {
                    ResourceName = "notification_icon",
                },
                Color =
                {
                    ResourceName = "colorPrimary"
                }
            },
            Description = "It's been 60 seconds since you set the notification",
            Schedule =
            {
                NotifyTime = DateTime.Now.AddSeconds(60),
            }
        };
        bool notificationSet = await NotificationCenter.Current.Show(notification);
        
  if (notificationSet)
        {
            await Shell.Current.DisplayAlert("Notification has been set",
                                "Notification set", "Ok");
        }
```
					
The alert / dialog confirms the result of NotificationCenter.Current.Show(notification); which is always true, even on the occasions prior to terminating the app when the notification doesn't show.

**Comment by sAMCodingProjects** [comment]
I am having the same problem, any updates?

**Comment by michaelconnolly1** [comment]
I have been using the plugin successfully for a couple of years but now I find that after I set up a local notification, say for tomorrow, and then close the app, the notification comes through. However, if I try the same thing again, no notifications come through. I have tried different versions of the plugin from version 4.1.4 upwards and get the same issue. This is only an issue when the app is closed. Notifications come through when the app is open or the app is in the background (but not closed).

I have also tried creating an apk and installing that rather than using Visual Studio to run the app but that hasn't make any difference.

I have tried used the sample provided and that works fine even when app is closed. However, if I modify the sample and add some additional code and rebuild, the notifications won't come through when the app is closed. I have distributed the apk to other people to try with different phones and they get the same issue (no notifications when app is closed).

Can anyone help, I've been at this for days and can't find why the notifications aren't coming through when the app is closed.

**Comment by thudugala** [comment]
@c0ff33-b34n Can you please attach a sample application

**Comment by c0ff33-b34n** [comment]
Hi @thudugala I have resolved the issue with notifications not appearing on first run when the app is terminated. It seems that it was caused by being in debug mode / being connected to Visual Studio. If I disconnect the cable to my mobile as soon as the app is installed before setting a notification and terminating it, or if I install a compiled apk directly on a device the notifications still work after termination of the app on first run.

I am still having problems with notifications not coming through on/after reboot though.
I've upgraded to the latest plugin version 9.1.3 and am now setting 
`AlarmType = Plugin.LocalNotification.AndroidOption.AndroidAlarmType.RtcWakeup` in Android options when creating the notification.

According to the official Android documentation on Alarms, ElapsedRealTime, as used in previous versions of the plugin, only gives a time since the device is booted, so I think I need to set the alarm type as one of the Rtc options for the timing to be correct on reboot.

I'm slightly confused by the BootReceiver implementation in the plugin. Am I right in understanding that it only recreates (reschedules) notifications that are repeating notifications? (As it only returns a list of those where `IsStillActiveForReSchedule()` is called and return `true` for` Schedule.RepeatType != NotificationRepeat.No`)

This would mean that one-off notifications in the future are not recreated after boot. Is that correct? How would I go about overcoming that / Is there a way to override this?

Secondly, if the time of a notification occurred while the device is off there appears to be no option to allow that notification to be retrieved/sent as soon as the device is switched on. Is that right?

**Comment by thudugala** [comment]
@c0ff33-b34n Yes you are right only IsStillActiveForReSchedule get set back in, This maybe wrong. 

> Secondly, if the time of a notification occurred while the device is off there appears to be no option to allow that notification to be retrieved/sent as soon as the device is switched on. Is that right?

Yes

Please do a PR to review suggested change.

Thank you for you testing results

**Comment by c0ff33-b34n** [comment]
@thudugala I will close this current issue now, as I discovered it appears to be a curiosity of using Visual Studio / debugging rather than the plugin code, but one to be mindful of while debugging.

I will raise separate issues for the one-off notifications not coming through after re-boot and the possibility of receiving notifications that have passed during the time the device was off if they are still within the AndroidAllowedDelay window.

I found another issue with BootReceiver, #251 and have proposed a change to fix it which I've tested multiple times with my device, which seems to resolve that one. Fixing the issue within the BootReceiver itself was better because firstly it doesn't affect the common code (affecting iOS) but also putting the check within GetNextNotifyTime() of NotificationRequest seemed to impact normal repeating notifications when I tested that alternative.
